odoo/addons/base/models/ir_mail_server.py

```diff
@@ -95,13 +100,28 @@ class IrMailServer(models.Model):
                                             "- None: SMTP sessions are done in cleartext.\n"
                                             "- TLS (STARTTLS): TLS encryption is requested at start of SMTP session (Recommended)\n"
                                             "- SSL/TLS: SMTP sessions are encrypted with SSL/TLS through a dedicated port (default: 465)")
+    smtp_ssl_certificate = fields.Binary('SSL Certificate', help='SSL certificate used for authentication')
+    smtp_ssl_private_key = fields.Binary('SSL Private Key', help='SSL private key used for authentication')
```

Comment:
I think you should set attachment=false and add a group on those fields. Maybe @odony want something else.

Reply:
It's true that we could be consistent with what is done for e.g. `smtp_pass`. Even if the model itself is only accessible to `group_system` *by default*, it doesn't hurt to add a group restriction on those ssl fields. Or we could remove the groups on all fields, and trust the admin won't make a mess with ACLs, as we do elsewhere :shrug: 

Regarding the attachment storage, `ir.attachment` does protect against reading the records that contain field values (`res_field` is set), so that should be ok. Yet those cert files are typically small, so it could be interesting to store them in-row, as that saves a few extra queries and FS access to read the value. Probably won't make a measurable difference either way, your call... ;-)